app/services/record_declarations/base.rb

```diff
@@ -10,9 +10,12 @@ class Base
     validates :declaration_type, inclusion: { in: :valid_declaration_types, message: I18n.t(:invalid_declaration_type) }
     validates :course_identifier, presence: { message: I18n.t(:missing_course_identifier) }
     validates :declaration_date, presence: { message: I18n.t(:missing_declaration_date) }
+    validates :declaration_date, declaration_date: true
+    validates :declaration_date, future_date: true
```

Comment:
Do we want to `allow_blank: true` for both of them? So that the we only catch the blank case once?